lib/connection.js

```diff
@@ -442,7 +442,7 @@ Connection.prototype.resume = function resume () {
 
 Connection.prototype.keyFromFields = function keyFromFields (fields, options) {
   var res = (typeof options.nestTables) + '/' + options.nestTables + '/' + options.rowsAsArray
-    + options.supportBigNumbers + '/' + options.bigNumberStrings;
+    + options.supportBigNumbers + '/' + options.bigNumberStrings + '/' + typeof options.typeCast;
```

Comment:
function itself is passed as a parameter, here I'm avoiding extra if/else in the parser if function is not callable at all